rental.rb

```diff
@@ -0,0 +1,19 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @book = book
+    @person = person
+    @date = date
+  end
+
+  def books=(book)
+    @book = book
+    book.rentals.push(self) unless book.rentals.include?(self)
+  end
+
+  def persons=(person)
+    @person = person
+    person.rentals.push(self) unless person.rentals.include?(self)
+  end
```

Comment:
[OPTIONAL]

- A better way of achieving this is to move all this in the constructor above as the conditions must be met because we've chained all in establishing all the relationships. So, therefore updating the constructor like this below:

```ruby


  def initialize(date, person, book)
    @date = date
    @person = person
    @book = book
    person.rentals << self
    book.rentals << self
  end
```

Reply:
Thank you. Changes implemented